google/cloud/bigtable/mutations_batcher.py

```diff
@@ -46,61 +174,221 @@ class MutationsBatcher:
+            self._flush_timer(flush_interval)
+        )
+        # create noop previous flush task to avoid None checks
+        self._prev_flush: asyncio.Task[None] = asyncio.create_task(asyncio.sleep(0))
+        # MutationExceptionGroup reports number of successful entries along with failures
+        self._entries_processed_since_last_raise: int = 0
+
+    async def _flush_timer(self, interval: float | None):
+        """
+        Triggers new flush tasks every `interval` seconds
+        """
+        if interval is None:
+            return
+        while not self.closed:
+            await asyncio.sleep(interval)
+            # add new flush task to list
+            if not self.closed and self._staged_mutations:
+                self._schedule_flush()
+
+    def append(self, mutations: RowMutationEntry | list[RowMutationEntry]):
+        """
+        Add a new set of mutations to the internal queue
 
-    async def append(self, row_key: str | bytes, mutation: Mutation | list[Mutation]):
+        Args:
+          - mutations: entries to add to flush queue
+        Raises:
+          - RuntimeError if batcher is closed
+          - ValueError if an invalid mutation type is added
         """
-        Add a new mutation to the internal queue
+        if self.closed:
+            raise RuntimeError("Cannot append to closed MutationsBatcher")
+        if not isinstance(mutations, list):
+            mutations = [mutations]
+        for m in mutations:
+            if isinstance(m, Mutation):  # type: ignore
+                raise ValueError(
+                    f"invalid mutation type: {type(m).__name__}. Only RowMutationEntry objects are supported by batcher"
+                )
+        self._staged_mutations.extend(mutations)
+        # start a new flush task if limits exceeded
+        self._staged_count += sum([len(m.mutations) for m in mutations])
+        self._staged_bytes += sum([m.size() for m in mutations])
+        if (
+            self._staged_count >= self._flush_limit_count
+            or self._staged_bytes >= self._flush_limit_bytes
+        ):
+            self._schedule_flush()
+
+    async def flush(self, *, raise_exceptions: bool = True, timeout: float | None = 60):
         """
-        raise NotImplementedError
+        Flush all staged mutations
 
-    async def append_conditional(
+        Args:
+          - raise_exceptions: if True, will raise any unreported exceptions from this or previous flushes.
+              If False, exceptions will be stored in self.exceptions and raised on a future flush
+              or when the batcher is closed.
+          - timeout: maximum time to wait for flush to complete, in seconds.
+              If exceeded, flush will continue in the background and exceptions
+              will be surfaced on the next flush
+        Raises:
+          - MutationsExceptionGroup if raise_exceptions is True and any mutations fail
```

Comment:
I dont we want anyone to ever disable exceptions here

Reply:
To be clear, any remaining exceptions will always be raise when the batcher is closed. This flag just allows users to create silent flush tasks that save their exceptions to be raised later, if they want to flush in the background somewhere.

I'm fine with removing it if that's what you want, but I do think there is a risk that some exceptions could be lost if background tasks aren't finalized properly